pkg/expression/builtin_time_test.go

```diff
@@ -41,6 +41,7 @@ import (
 
 func TestDate(t *testing.T) {
 	ctx := createContext(t)
+	ctx.GetSessionVars().SQLMode = mysql.DelSQLMode(ctx.GetSessionVars().SQLMode, mysql.ModeNoZeroDate)
```

Comment:
Why modify this

Reply:
The `mock.NewContext` returns a context with an empty sql mode but set `StrictSQLMode` to true in the previous implement. So in the current implementation we set SQL mode to the default `DefaultSQLMode` to make it a real strict mode to make some UT pass. 

`DefaultSQLMode` contains `NO_ZERO_DATE` date, it will cause some cases fail in this test (this test suppose `NO_ZERO_DATE` not set).

Reply:
But I think this line can be moved before: https://github.com/pingcap/tidb/pull/50624/files/8dc99b08aa6d569da36fc734aea6c87ff01aa88c#diff-210d7ae9ffb6cec2ad0af174f954bfcfa98ed15aa3a5efe2b831dfcefb0e20d2R231